settings.go

```diff
@@ -75,6 +85,34 @@ func SetMwindowMappedLimit(size int) error {
 	return setSizet(C.GIT_OPT_SET_MWINDOW_MAPPED_LIMIT, size)
 }
 
+func EnableCaching(enabled bool) error {
+	       if enabled {
+		               return setSizet(C.GIT_OPT_ENABLE_CACHING, 1)
+		       } else {
+		               return setSizet(C.GIT_OPT_ENABLE_CACHING, 0)
+		       }
+}
+
+func GetCachedMemory() (int, int, error) {
```

Comment:
```suggestion
func GetCachedMemory() (current int, allowed int, err error) {
```

this should make the names appear in godoc, saving a trip to the libgit2 docs.